### PR TITLE
feat: Kill last zm alive?

### PR DESCRIPTION
**Pull request by Rushaway**
Second check is for this case:

**2 Zombies alive ::** 

1. [SM] Rick knifed Derek
2. [SM] Rick knifed .Rushaway
3. [Knife Mode] You have 3.0 seconds to catch any human or you will die!
4. Rick killed Derek with knife.
_>>> only 1 zombie still alive
(Pack is already passed in the ByeZM timer -> second check needed to dont kill last zm)_

5. [Knife Mode] You are the last Zombie alive, canceling your die!

### CODE REVIEW

**Comment by maximee-leroyy** [comment]
so if i understand correctly its to have a mode so that we have to kill the last zm with weapons ?

File: addons/sourcemod/scripting/KnifeMode.sp [hunk 1]
```diff
@@ -175,12 +180,25 @@ public Action ZR_OnClientInfect(int &client, int &attacker, bool &motherInfect,
...
-    attacker = ReadPackCell(pack);
+    int client = ReadPackCell(pack);
+    int attacker = ReadPackCell(pack);
+
+    // Another check : In case 2 different pack is in progress for differents clients
+    if (g_cvKillLastZM.IntValue == 0 && GetTeamAliveCount(2) <= 1)
+    {
+        if (GetEngineVersion() == Engine_CSGO)
+        {
+            PrintHintText(client, "<font class='fontSize-l' color='#00ff00'>[Knife Mode]</font> <font class='fontSize-l'>You are the last Zombie alive, canceling your die!</font>");
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/KnifeMode.sp]: canceling your death

same thing for other sentences

File: addons/sourcemod/scripting/KnifeMode.sp [hunk 2]
```diff
@@ -38,6 +39,7 @@ public void OnPluginStart()
     g_cvExplodeTime = CreateConVar("sm_knifemode_time", "3", "Seconds that a zombie has to catch any human");
     g_cvUnload = CreateConVar("sm_knifemode_unload", "0", "Automaticaly unload plugin on map end [0 = No | 1 = Yes, unload it.]");
     g_cvSpectateDisable = CreateConVar("sm_knifemode_spectate_disable", "0", "Automaticaly disable the spectate plugin on map start [0 = No | 1 = Yes, disable it.]");
+    g_cvKillLastZM = CreateConVar("sm_knifemode_kill_lastzm", "0", "Allow last zombie alive to be killed by a knife ? [0 = No | 1 = Yes, kill it.]");
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/KnifeMode.sp]: default to 1

**Comment by Rushaway** [comment]
> so if i understand correctly its to have a mode so that we have to kill the last zm with weapons ?

Yes, but 99% of servers are allowing zombie respawn after they die, or they auto respawn as zm. 
This cvar is made for ZE server.

**Comment by maximee-leroyy** [comment]
yes i understand ok for this behavior on ze

**Pull request merged**